### PR TITLE
[m0][bgp] Add support for m0 in bgp test to simulate both t0 and t1 scenario

### PR DESCRIPTION
**Pull request by yaqiangz**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205

#### What is the motivation for this PR?
In current bgp test for m0, just simulate scenario that m0 device performs like t1. But m0 is like both t1 and t0.

#### How did you do it?
Add suport for m0 in bgp test to simulate both t0 and t1 scenario

#### How did you verify/test it?
Run tests on m0 topo and t0 topo
M0 topo:
```
bgp/test_bgp_update_timer.py::test_bgp_update_timer[testbed-m0_t0_scenario] PASSED                                                                                            [ 50%]
bgp/test_bgp_update_timer.py::test_bgp_update_timer[testbed-m0_t1_scenario] PASSED                                                                                            [100%]
```
T0 topo:
```
bgp/test_bgp_update_timer.py::test_bgp_update_timer[testbed-default] PASSED                                                                                            [100%]
```

### CODE REVIEW

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Passed<br/>fix end of files.........................................................Passed<br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/><br/>tests/bgp/conftest.py:253:121: E501 line too long (136 > 120 characters)<br/>tests/conftest.py:1714:1: E302 expected 2 blank lines, found 1<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

File: tests/conftest.py [hunk 1]
```diff
@@ -1417,6 +1417,12 @@ def format_portautoneg_test_id(param):
     if 'enum_pfc_pause_delay_test_params' in metafunc.fixturenames:
         metafunc.parametrize("enum_pfc_pause_delay_test_params", pfc_pause_delay_test_params(metafunc))
 
+    if 'topo_scenario' in metafunc.fixturenames:
+        if tbinfo['topo']['type'] == 'm0' and 'topo_scenario' in metafunc.fixturenames:
+            metafunc.parametrize('topo_scenario', ['m0_t0_scenario', 'm0_t1_scenario'], scope='module')
+        else:
+            metafunc.parametrize('topo_scenario', [None], scope='module')
```
Comment (Blueve) [inline tests/conftest.py:1424]: Suggest use `default`, `None` typically infers incomplete setup

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Passed<br/>fix end of files.........................................................Passed<br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/><br/>tests/bgp/conftest.py:253:121: E501 line too long (136 > 120 characters)<br/>tests/conftest.py:1714:1: E302 expected 2 blank lines, found 1<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

File: tests/conftest.py (hunk 1 above)
Comment (yaqiangz) [reply tests/conftest.py]: Fixed

**Review by Blueve — APPROVED** [review]

**Pull request merged**